src/generic/datavgen.cpp

```diff
@@ -2686,14 +2846,25 @@ bool wxDataViewMainWindow::ItemDeleted(const wxDataViewItem& parent,
 bool wxDataViewMainWindow::ItemChanged(const wxDataViewItem & item)
 {
     SortPrepare();
-    GetModel()->Resort();
+
+    // Instead of resorting the parent, just move this specific child node.
+    wxDataViewItem parentItem = GetModel()->GetParent(item);
+    wxDataViewTreeNode * parentNode = FindNode(parentItem);
+    if ( parentNode )
+    {
+        wxDataViewTreeNode * node = FindNode(item);
+        wxASSERT(node);
+        parentNode->MoveUpdatedChild(node);
+    }
 
     GetOwner()->InvalidateColBestWidths();
 
     // Send event
     wxDataViewEvent le(wxEVT_DATAVIEW_ITEM_VALUE_CHANGED, m_owner, item);
     m_owner->ProcessWindowEvent(le);
 
+	// Make sure the change is actually shown right away
+	UpdateDisplay();
```

Comment:
As someone looking into [noticeable performance problems even in *drawing code*](https://github.com/vslavik/poedit/issues/340), I’ll comment on this line: this, i.e. immediately redrawing *the entire control* on every change, would make already bad drawing performance even worse: even drawing the full control at idle time, as is currently done, is too slow.